File: android-app/app/src/main/java/com/example/fingerprintdemo/rest/Enrollment.java
package com.example.fingerprintdemo.rest;

public class Enrollment implements SignableRequestPayload {

    private final String username;
    private final String password;
    private final String deviceId;
    private final String publicKey;
    private final long timestamp;

    public Enrollment(String username, String password, String deviceId, String publicKey) {
        this.username = username;
        this.password = password;
        this.deviceId = deviceId;
        this.publicKey = publicKey;
        this.timestamp = System.currentTimeMillis();
    }

    @Override
    public String concatenatedValuesForSigning() {
        return username
                + "|"
                + password
                + "|"
                + deviceId
                + "|"
                + publicKey
                + "|"
                + timestamp;
    }
}

File: demo-backend/src/main/kotlin/com/example/demobackend/Enrollment.kt
package com.example.demobackend

class Enrollment : SignableRequestPayload {
    lateinit var username: String
    lateinit var password: String
    lateinit var deviceId: String
    lateinit var publicKey: String

    override var timestamp: Long = 0


    override fun messageForVerification(): String {
        return listOf(username,
                password,
                deviceId,
                publicKey,
                timestamp).joinToString("|")
    }
}


